eng/pipelines/templates/steps/build.yml

```diff
@@ -60,17 +60,12 @@ steps:
       Artifacts: ${{ parameters.Artifacts }}
       ArtifactPath: $(Build.SourcesDirectory)/sdk/${{ parameters.ServiceDirectory }}
 
-  # It's important for performance to pass "sdk" as "sourceFolder" rather than as a prefix in "contents".
-  # The task first enumerates all files under "sourceFolder", then matches them against the "contents" pattern.
-  - task: CopyFiles@2
-    inputs:
-      sourceFolder: sdk
-      contents: |
-        **/$(coalesceResultFilter)/*.tgz
-        **/$(coalesceResultFilter)/browser/*.zip
-      targetFolder: $(Build.ArtifactStagingDirectory)
-      flattenFolders: true
-    displayName: "Copy packages"
+  - ${{ each artifact in parameters.Artifacts }}:
+    - pwsh: |
+        New-Item -Type Directory -Name ${{artifact.name}} -Path $(Build.ArtifactStagingDirectory)
+        Copy-Item sdk/${{parameters.ServiceDirectory}}/**/${{artifact.name}}-*.tgz $(Build.ArtifactStagingDirectory)/${{artifact.name}}
+        Copy-Item sdk/${{parameters.ServiceDirectory}}/**/browser/${{artifact.name}}-*.zip $(Build.ArtifactStagingDirectory)/${{artifact.name}}
+      displayName: 'Copy Package ${{artifact.name}}'
 
   - template: /eng/common/pipelines/templates/steps/publish-artifact.yml
```

Comment:
Lets see if we can eliminate all the other publishing accept for the packages artifact by combining the contents into the packages artifact.